src/lib/helpers/uploader.ts

```diff
@@ -0,0 +1,47 @@
+import imageCompression from 'browser-image-compression';
+import { keccak256 } from 'ethers/lib/utils';
+import { AddResult } from 'ipfs-core-types/src/root';
+import { ImportCandidate } from 'ipfs-core-types/src/utils';
+import { create, Options } from 'ipfs-http-client';
+
+import { LSP3ProfileImage } from '../interfaces';
+import { ProfileUploadOptions } from '../interfaces/profile-upload-options';
+
+export const sizes = [1800, 1024, 640, 320, 180];
+export async function imageUpload(
+  givenFile: File,
+  uploadOptions: ProfileUploadOptions
+): Promise<LSP3ProfileImage[]> {
+  const isImage = givenFile.type?.substr(0, 6) === 'image/';
+  if (!isImage) {
+    throw new Error('File provided is not an image.');
```

Comment:
Maybe to help debug you can add in the log "given file type is ${givenFile.type}" ?

Reply:
Done.